api/app/src/main/kotlin/packit/service/UserService.kt

```diff
@@ -80,7 +83,11 @@ class BaseUserService(
     override fun getUserForLogin(username: String): User
     {
         val user =
-            userRepository.findByUsername(username) ?: throw PackitException("userNotFound", HttpStatus.UNAUTHORIZED)
+            userRepository.findByUsername(username) ?: throw AuthenticationException()
+        if (user.lastLoggedIn == null)
+        {
+            throw AuthenticationException("You must change your password before logging in.")
+        }
```

Comment:
Sorry I haven't actually run this up, but I think this means that even if the user puts in the wrong password on the initial login screen, they'll still get redirected to the change password page here, because the password checking happens after this is called (?). That's not catastrophic as they will still need to get their password correct in order to change it, but it does feel like a wrong password should be flagged immediately before getting handed to the next screen. User might interpret this behaviour as meaning that they did get their pw right the first time, and be confused when they can't complete the update password form with the same pw...

Reply:
yip 100% right... initially was because behind spring auth.. but I just moved after the auth check..